File: MPI/mpicheck/tests/functions/3.0/MPI_Imrecv.f90

        
        program main
        use mpi
        
        TYPE(INTEGER) var_0(10)
       INTEGER var_1
       INTEGER var_2
       INTEGER var_3
       INTEGER var_4
       INTEGER var_5
        call mpi_imrecv(var_0, var_1, var_2, var_3, var_4, var_5)
       call pmpi_imrecv(var_0, var_1, var_2, var_3, var_4, var_5)
        end program main
    

File: MPI/mpicheck/tests/functions/3.0/MPI_Imrecv.c
#include <mpi.h>
int main(char argc, char**argv)
{
    /* vars */
    void *var_0;
    int var_1;
    MPI_Datatype var_2;
    MPI_Message *var_3;
    MPI_Request *var_4;
    int ret;
    /* calls */
    ret = MPI_Imrecv(var_0, var_1, var_2, var_3, var_4);
    ret = PMPI_Imrecv(var_0, var_1, var_2, var_3, var_4);
    return 0;
}